mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -190,7 +210,8 @@ static void overlayTableProperties(Configuration configuration, TableDesc tableD
     map.put(InputFormatConfig.TABLE_SCHEMA, schemaJson);
 
     if (table instanceof Serializable) {
-      map.put(InputFormatConfig.SERIALIZED_TABLE, SerializationUtil.serializeToBase64(table));
+      map.put(InputFormatConfig.SERIALIZED_TABLE_PREFIX + tableDesc.getTableName(),
```

Comment:
Once we have the table-level catalog config merged, will it be enough to use the tableName here? E.g. if we write to `db.tbl` from Catalog A and `db.tbl` from Catalog B in the same query, won't they have the same value for `tableDesc.getTableName()` and therefore the second one will overwrite the first one?

Reply:
The `tableName` is the unique name from the HMS tableName. The Iceberg `table.name` is the one which could be duplicated it the same name is used in different Iceberg Catalogs.

Reply:
Sounds good then